examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairingCommand.kt

```diff
@@ -0,0 +1,235 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package com.matter.controller.commands.pairing
+
+import chip.devicecontroller.ChipDeviceController
+import com.matter.controller.commands.common.CredentialsIssuer
+import com.matter.controller.commands.common.IPAddress
+import com.matter.controller.commands.common.MatterCommand
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+import java.util.logging.Level
+import java.util.logging.Logger
+
+abstract class PairingCommand(
+  controller: ChipDeviceController,
+  commandName: String?,
+  credsIssuer: CredentialsIssuer?,
+  private var pairingMode: PairingModeType = PairingModeType.NONE,
+  private var networkType: PairingNetworkType = PairingNetworkType.NONE,
+  private var filterType: DiscoveryFilterType = DiscoveryFilterType.NONE
+) : MatterCommand(controller, commandName, credsIssuer), ChipDeviceController.CompletionListener {
+  private var remoteAddr: IPAddress? = null
```

Comment:
This can be a `val` and of type `IPAddress` (it can never be null).

```
private val remoteAddr: IPAddress = IPAddress(InetAddress.getByName("::1"))
```

You can also initialize it the same way in the `init {}` block if you prefer (as a `val`) but it's preferable to declare it inline:

```
private val remoteAddr: IPAddress

init {
  remoteAddr = IPAddress(InetAddress.getByName("::1"))
}
```